src/layout/layout.rs

```diff
@@ -572,16 +559,11 @@ impl Layout {
     /// ```
     pub fn split_with_spacers(&self, area: Rect) -> (Segments, Spacers) {
         LAYOUT_CACHE.with(|c| {
-            c.get_or_init(|| {
-                RefCell::new(LruCache::new(
-                    NonZeroUsize::new(Self::DEFAULT_CACHE_SIZE).unwrap(),
-                ))
-            })
-            .borrow_mut()
-            .get_or_insert((area, self.clone()), || {
-                self.try_split(area).expect("failed to split")
-            })
-            .clone()
+            c.borrow_mut()
+                .get_or_insert((area, self.clone()), || {
```

Comment:
I'd extract (area, self.clone() into a variable named key. You end up with more expressive code and even save a line as the block no longer wraps.